src/Microsoft.Graph.Core/Extensions/BaseRequestExtensions.cs

```diff
@@ -167,5 +167,41 @@ public static class BaseRequestExtensions
 
             return baseRequest;
         }
+
+        /// <summary>
+        /// Sets Microsoft Graph's scopes that will be used by <see cref="IAuthenticationProvider"/> to authenticate this request
+        /// and can be used to perform incremental scope consent.
+        /// This only works with the default authentication handler and default set of Microsoft graph authentication providers.
+        /// If you use a custom authentication handler or authentication provider, you have to handle it's retrieval in your implementation.
+        /// </summary>
+        /// <param name="baseRequest">The <see cref="IBaseRequest"/>.</param>
+        /// <param name="scopes">Microsoft graph scopes used to authenticate this request.</param>
+        public static T WithScopes<T>(this T baseRequest, string[] scopes) where T : IBaseRequest
+        {
+            string authHandlerOptionKey = typeof(AuthenticationHandlerOption).ToString();
```

Comment:
nameof instead?

Reply:
unresolving this one, it looks like it's been missed

Reply:
@baywet I am hesitant to change this as the style used in the rest of the codebase to insert key values in the `MiddlewareOptions` collection is `typeof(AuthenticationHandlerOption).ToString()` which results to the string `AuthenticationHandlerOption` whereas `nameof(AuthenticationHandlerOption)` results in `Microsoft.Graph.AuthenticationHandlerOption`. A few examples [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/e7ab26c8863b9ab0ca450ca95959d231205fad6a/src/Microsoft.Graph.Core/Extensions/BaseRequestExtensions.cs#L26), [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/e7ab26c8863b9ab0ca450ca95959d231205fad6a/src/Microsoft.Graph.Core/Extensions/BaseRequestExtensions.cs#L51).

If I decide to change one, we could potentially cause a mismatch when reading from the `MiddlewareOptions` collection if we do not change them all correctly.

Reply:
besides having to change it at other places, what would be the other negative impact potentially?

I'm worried about potential performance impacts where typeof does a complete reflection on the type but nameof only gets the fully qualified class name. Besides, I believe it should be typeof(Class).Name instead of ToString. In theory you never know when the ToString representation of the object will change (in practice it's unlikely they'll change it bit fear of breaking people's apps)

Reply:
also, from what I can tell, nameof returns the class short name, not the fully qualified name https://dotnetfiddle.net/JIpBfi

Reply:
> typeof(AuthenticationHandlerOption).ToString() which results to the string AuthenticationHandlerOption whereas nameof(AuthenticationHandlerOption) results in Microsoft.Graph.AuthenticationHandlerOption

Sorry about the confusion. This is actually the other way round. 

> I'm worried about potential performance impacts where typeof does a complete reflection on the type but nameof only gets the fully qualified class name.

Now I definitely have to fix this 😅. Since it will be in a separate PR, it will be much easier to track the specific change in case we miss something.

Reply:
to provide hard evidence: nameof is just [syntax sugar and gets translated at compile time](https://www.erikschierboom.com/2015/12/31/csharp6-under-the-hood-nameof/), at runtime, it's just a constant string.
[typeof calls a bunch of methods under the hood at runtime](https://stackoverflow.com/questions/6417763/efficiency-of-cs-typeof-operator-or-whatever-its-representation-is-in-msil/6417977#6417977).
So it's always going to be faster to use nameof compared to typeof + Name.

Reply:
@baywet Unfortunately, after attempting to do the refactoring, I don't think we can use `nameof` because of the function [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/e7ab26c8863b9ab0ca450ca95959d231205fad6a/src/Microsoft.Graph.Core/Extensions/HttpRequestMessageExtensions.cs#L120). 😞 

Since we are using generics in that case to retrieve information from the `MiddlewareOptions` collection, `nameof` returns the string `T` rather than the key value that we would want. This is also covered in the restrictions section of your link earlier [here](https://www.erikschierboom.com/2015/12/31/csharp6-under-the-hood-nameof/#restrictions).

Reply:
keep the typeof at that place, but read the Name property (shortname) instead of the ToString() return value, it should work :)

Reply:
Thanks @baywet! This is now resolved via #275